### PR TITLE
Add --benchmark_list_tests option and change filter to match generated name.

### PR DESCRIPTION
**Pull request by EricWF**

### CODE REVIEW

File: src/benchmark.cc [hunk 1]
```diff
@@ -414,7 +418,8 @@ bool BenchmarkFamilies::FindBenchmarks(
           instance.name += StringPrintF("/threads:%d", instance.threads);
         }
 
-        benchmarks->push_back(instance);
+        if (re.Match(instance.name))
```
Comment (dominichamon) [inline src/benchmark.cc]: it's not required, but for consistency these should have {} even though they're single-line.

File: src/benchmark.cc [hunk 2]
```diff
@@ -766,19 +771,29 @@ void State::SetLabel(const char* label) {
...
 
 namespace internal {
+namespace {
+
+void PrintBenchmarkList() {
+  std::vector<Benchmark::Instance> benchmarks;
+  auto families = BenchmarkFamilies::GetInstance();
+  if (!families->FindBenchmarks(".", &benchmarks)) return;
+
+  for (const internal::Benchmark::Instance& benchmark : benchmarks)
```
Comment (dominichamon) [inline src/benchmark.cc]: {} please

File: src/benchmark.cc (hunk 1 above)
Comment (EricWF) [inline src/benchmark.cc]: ack.

File: src/benchmark.cc (hunk 2 above)
Comment (EricWF) [inline src/benchmark.cc]: ack.

**Pull request merged**